yarn-project/pxe/src/bin/check_oracle_version.ts

```diff
@@ -20,18 +16,35 @@ class OracleMock extends TypedOracle {}
  * oracle interface like we do in PXE (i.e., there is no single Oracle class that contains only the oracles).
  */
 function assertOracleInterfaceMatches(): void {
-  const oracle = new Oracle(new OracleMock('OracleMock'));
-  const acirCallback = oracle.toACIRCallback();
+  const excludedProps = [
+    'handler',
+    'constructor',
+    'toACIRCallback',
+    'handlerAsMisc',
+    'handlerAsUtility',
+    'handlerAsPrivate',
+  ] as const;
+
+  // Get all the oracle function names
+  // TODO: we're only checking the functions implemented by the Oracle object, which is really an ACVM translator, akin
+  // to TXE's RPC translator. This is correct in that it is ultimately the foreign interface that matters, but incorrect
+  // in that it does not take into consideration TXE's extended interface. An improvement would be to check
+  // RPCTranslator from TXE, though that still seems a bit fragile.
```

Comment:
Would merge the comment on line 14 with this one and keep the link to the issue "TODO(#16581)".